aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -107,10 +105,16 @@
         // force a new connection to be created.
         if (webSocketListener == null || webSocketListener.isDisconnectedState()) {
             webSocketListener = new AmplifyWebSocketListener();
-            webSocket = okHttpClient.newWebSocket(new Request.Builder()
-                .url(subscriptionUrl)
-                .addHeader("Sec-WebSocket-Protocol", "graphql-ws")
-                .build(), webSocketListener);
+            try {
+                webSocket = okHttpClient.newWebSocket(new Request.Builder()
+                    .url(buildConnectionRequestUrl())
+                    .addHeader("Sec-WebSocket-Protocol", "graphql-ws")
+                    .build(), webSocketListener);
+            } catch (ApiException apiException) {
```

Comment:
What actually throws the `ApiException`? The `buildConnectionRequestUrl()` bit? The rest looks likes its pure OkHttp, and wouldn't throw one of our internal types.

Reply:
Yeah...it's buried deep down into the code that adds the security headers.